libbeat/processors/decode_xml/decode_xml_test.go

```diff
@@ -0,0 +1,482 @@
+						<title>Midnight Rain</title>
+						<review>Some review.</review>
+					</book>
+				</catalog>`,
+			},
+			Output: common.MapStr{
+				"message": map[string]interface{}{
+					"catalog": map[string]interface{}{
+						"book": []interface{}{
+							map[string]interface{}{
+								"author": "William H. Gaddis",
+								"review": "One of the great seminal American novels of the 20th century.",
+								"title":  "The Recognitions",
+							},
+							map[string]interface{}{
+								"author": "Ralls, Kim",
+								"review": "Some review.",
+								"title":  "Midnight Rain",
+							},
+						},
+					},
+				},
+			},
+			error:        false,
+			errorMessage: "",
+		},
+		{
+			description: "Decoding with multiple xml objects",
+			config: decodeXMLConfig{
+				Field: "message",
+			},
+			Input: common.MapStr{
+				"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+					<author>William H. Gaddis</author>
+					<title>The Recognitions</title>
+					<review>One of the great seminal American novels of the 20th century.</review>
+				</book>
+				<book>
+					<author>Ralls, Kim</author>
+					<title>Midnight Rain</title>
+					<review>Some review.</review>
+				</book>
+				<secondcategory>
+					<paper id="bk102">
+						<test2>Ralls, Kim</test2>
+						<description>A former architect battles corporate zombies, 
+						an evil sorceress, and her own childhood to become queen 
+						of the world.</description>
+					</paper>
+				</secondcategory>
+				</catalog>`,
+			},
+			Output: common.MapStr{
+				"message": map[string]interface{}{
+					"catalog": map[string]interface{}{
+						"book": []interface{}{
+							map[string]interface{}{
+								"author": "William H. Gaddis",
+								"review": "One of the great seminal American novels of the 20th century.",
+								"title":  "The Recognitions",
+							},
+							map[string]interface{}{
+								"author": "Ralls, Kim",
+								"review": "Some review.",
+								"title":  "Midnight Rain",
+							},
+						},
+						"secondcategory": map[string]interface{}{
+							"paper": map[string]interface{}{
+								"description": "A former architect battles corporate zombies, \n\t\t\t\t\t\tan evil sorceress, and her own childhood to become queen \n\t\t\t\t\t\tof the world.",
+								"id":          "bk102",
+								"test2":       "Ralls, Kim",
+							},
+						},
+					},
+				},
+			},
+			error:        false,
+			errorMessage: "",
+		},
+		{
+			description: "Decoding with broken XML format, with IgnoreFailure false",
+			config: decodeXMLConfig{
+				Field:         "message",
+				IgnoreFailure: false,
+			},
+			Input: common.MapStr{
+				"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+						<author>William H. Gaddis</author>
+						<title>The Recognitions</title>
+						<review>One of the great seminal American novels of the 20th century.</review>
+				</ook>
+				catalog>`,
+			},
+			Output: common.MapStr{
+				"message": (map[string]interface{})(nil),
```

Comment:
Where did `message` go? It failed and it deleted the source data. That seems bad.